docs/styling-with-css/index.md

```diff
@@ -0,0 +1,338 @@
+```
+
+The `{||}` string literal is known as a [quoted
+string literal](https://v2.ocaml.org/manual/lex.html#sss:stringliterals), and it
+is used to represent strings of arbitrary content without escaping[^3]. They are
+similar to the `{js||js}` string literals we first saw in the [Celsius
+Converter](/celsius-converter-exception/#solutions) chapter, with the difference
+that they won't handle Unicode correctly.
+
+Unfortunately, in the terminal where we're running `make serve`, we see this
+Webpack compilation error:
+
+```
+ERROR in ./_build/default/src/order-confirmation/output/src/order-confirmation/Order.js 7:0-8:1
+Module not found: Error: Can't resolve './order-item.css' in '~/melange-for-react-devs/_build/default/src/order-confirmation/output/src/order-confirmation'
+```
+
+## Tell Dune to copy CSS files
+
+The problem is that Webpack is serving the app from the build directory at
+`_build/default/src/order-confirmation/output/src/order-confirmation`, and the
+`order-item.css` file isn't in that build directory.
+
+To solve this, we can add the [runtime_deps
+field](https://melange.re/v2.1.0/build-system/#handling-assets) to our
+`melange.emit` stanza in `src/order-confirmation/dune`:
+
+```clj{7}
+(melange.emit
+ (target output)
+ (libraries reason-react)
+ (preprocess
+  (pps melange.ppx reason-react-ppx))
+ (module_systems es6)
+ (runtime_deps order-item.css))
+```
+
+We also want to add styles for the `Order` component, so add a new file
+`src/order-confirmation/order.css` with these styles:
+
+<!-- This file gets renamed to order.module.css later -->
+<<< @/../src/styling-with-css/order.module.css
+
+To ensure that `order.css` is also copied to the build directory, we can add
+`order.css` to the value of `runtime_deps`:
+
+```clj
+(runtime_deps item.css order.css)
+```
+
+If you have many `.css` files, you can tell `runtime_deps` to copy all `.css`
+files over using Dune's `glob_files` configuration:
+
+```clj
+(runtime_deps (glob_files *.css))
+```
+
+Check the Dune documentation for the [different options for
+globs](https://dune.readthedocs.io/en/stable/concepts/dependency-spec.html#glob).
+
+## Add classes to JSX
+
+Now we can add the appropriate classes to `OrderItem.make`'s JSX:
+
+<<< Order.re#order-item{4-6}
+
+As well as `Order.make`'s JSX:
+
+<<< Order.re#order-make{6,13}
+
+Finally, add a `mel.raw` extension node at the top of `Order.re`:
+
+```reason
+[%%mel.raw {|import "./order.css"|}];
+```
+
+## Problems with `mel.raw`
+
+This solution works well for our current build configuration, but falls apart if
+we change the `module_systems` field of the `melange.emit` stanza from `es6` to
+`commonjs`. This results in a subtle runtime error caused by the fact
+that CommonJS uses `require` instead of `import` to import modules.
+
+The `mel.raw` extension node is unsafe, but it is still useful for prototyping.
+Fortunately, Melange provides an official and more reliable way to import
+frontend assets.
```

Comment:
Also, not sure if we should elaborate more. `external` is not strictly to import frontend assets, but more generically, allows to interact between OCaml code and any piece of existing JavaScript code 🤔 

Edit: oh, I see it's mentioned below:

https://github.com/melange-re/melange-for-react-devs/blob/096fb4c897d36c853ee6bb75715e1a47dae8c85e/docs/styling-with-css/index.md?plain=1#L228-L231